### DISCUSSION TITLE
How can i apply CRUD operations in Semantic Kernel memory?

### DISCUSSION THREAD

**Discussion by simone-devoti** [microsoft/semantic-kernel#2143] [Q&A]
I am writing a chatbot that can be trained with custom data.
I would like to give the user one page where he can add/delete/update or view all of the elements inside the Semantic Kernel memory

The available methods for ISemanticTextMemory are:
- SaveInformationAsync => I can use this for Create/Update
- SaveReferenceAsync
- GetAsync
- RemoveAsync => I can use this for Delete
- SearchAsync
- GetCollections

The missing methods in order to implement all crud operations in my opinion are:
- GetAllAsync
- GetRangeAsync(int startIndex, int endIndex) => Some method to get elements with pagination

I can implement the CRUD operations using another database and connect the two but it seems like to be unnecessary work

Any ideas on how to do CRUD operations?

Thanks for your time
Simone

**Comment by MSFTJim** [comment]
I too am looking for a GetAll (points or memories) in a Collection.  It would be helpful to also get support for the count capability as well.

Currently I am using qdrant and getting the information via its API.
![image](https://github.com/microsoft/semantic-kernel/assets/33520581/efb809f1-c79c-4ebe-8bcd-f3b7837d4a34)

**Reply by timrobertsusa** [reply]
**looking for same**-- and also-- have an "green" ongoing list of which models are supported -- there are differences on what the underlying model/orchestrator supports. I am using qdrant for now but would like to explore others in the vector DB offerings.

**Comment by simone-devoti** [comment]
If you want to follow, I opened a feature request here #2237

**Comment by markwallace-microsoft** [comment]
I am cleaning up old unanswered discussions
